### PR TITLE
Backend changes to detect noauthsessionservice for e2e group tests

### PR DESCRIPTION
**Pull request by dianab0**
Added necessary changes to backend SessionServiceController in order to have the e2e tests for groups working when noauthsessionservice is used.

This way the /group/id and /groups/fetch controllers will work when noauthsessionservice is used instead of returning error 503.

### CODE REVIEW

**Comment by dianab0** [comment]
Hi, I just pushed the changes from review. Please check.

**Comment by jjgao** [comment]
@pvannierop what would be the next step for this pr?

**Comment by pvannierop** [comment]
@jjgao All is good to go. Diana forgot to answer the questions in text :)

**Review by pvannierop — APPROVED** [review]
All is good now

**Comment by jjgao** [comment]
Thanks, @pvannierop. @kalletlak could you also help review this pr? Thanks

File: web/src/main/java/org/cbioportal/web/SessionServiceController.java [hunk 1]
```diff
@@ -297,10 +307,9 @@ public void updateUsersInVirtualStudy(@PathVariable SessionType type, @PathVaria
     public ResponseEntity<List<VirtualStudy>> fetchUserGroups(
             @Size(min = 1, max = PagingConstants.MAX_PAGE_SIZE) @RequestBody List<String> studyIds,
             HttpServletResponse response) throws IOException {
+        if ((isSessionServiceEnabled() && isAuthorized()) || isNoAuthSessionServiceActive()) {
+            String username = isNoAuthSessionServiceActive() ? "anonymous" : userName();
```
Comment (kalletlak) [inline web/src/main/java/org/cbioportal/web/SessionServiceController.java:311]: default username is `anonymous` for `social_auth_google` and `social_auth_microsoft` authenticate types. Using this for `noauthsessionservice` would fetch all groups created with the those authenticate modes.

File: web/src/main/java/org/cbioportal/web/SessionServiceController.java [hunk 2]
```diff
@@ -370,35 +379,44 @@ public void updateUserPageSettings(@RequestBody PageSettingsData settingsData, H
...
-
             if (session.getType().equals(SessionType.group)) {
 
                 VirtualStudy virtualStudy = mapper.readValue(mapper.writeValueAsString(session), VirtualStudy.class);
 
                 VirtualStudyData virtualStudyData = virtualStudy.getData();
+
                 // only allow owner to update virtual study
-                if (userName().equals(virtualStudyData.getOwner()) && virtualStudy.getType().equals(type)) {
+                String username = isNoAuthSessionServiceActive() ? "anonymous" : userName();
```
Comment (kalletlak) [inline web/src/main/java/org/cbioportal/web/SessionServiceController.java:395]: Even here

**Review by kalletlak — CHANGES_REQUESTED** [review]
Looks like some of the changes are related to update session(update group), I forgot to remove `group` from updating. do we still want it?

cc @jjgao

File: web/src/main/java/org/cbioportal/web/SessionServiceController.java (hunk 1 above)
Comment (pvannierop) [reply web/src/main/java/org/cbioportal/web/SessionServiceController.java:311]: @kalletlak The goal here is to have session service to work when locally testing w/o authentication getting in the way. I have no other concerns and, due to limited/no understanding of session service, I cannot see beyond the current proposa.. Do you have a recommendation here?

**Comment by jjgao** [comment]
@kalletlak @pvannierop any comments on this?

**Comment by pvannierop** [comment]
@kalletlak @jjgao We will return our attention to this feature soon. Thanks for reminding us on this PR.

**Comment by pvannierop** [comment]
No longer needed now that keycloak will be included in e2e-localdb tests.

**Pull request closed without merging**